app/src/main/kotlin/it/vfsfitvnm/vimusic/download/MediaDownloadService.kt

```diff
@@ -30,13 +32,29 @@ class MediaDownloadService : DownloadService(DOWNLOAD_NOTIFICATION_ID) {
 
         val dataSourceFactory = DefaultHttpDataSource.Factory()
 
-        return DownloadManager(
+        val downloadManager = DownloadManager(
             this,
             databaseProvider,
             globalCache,
             dataSourceFactory,
             executor
         )
+        downloadManager.maxParallelDownloads = 9999
+        downloadManager.addListener(object : DownloadManager.Listener {
+            override fun onDownloadChanged(
+                downloadManager: DownloadManager,
+                download: Download,
+                finalException: Exception?
+            ) {
+                if(download.bytesDownloaded == download.contentLength) {
+                    query {
+                        Database.markDownloaded()
```

Comment:
why do all songs in database have isDownloaded true?

Reply:
Because if we download song we mark it as downloaded, so if song saved in database, it has isDownload true. We don't store songs with isDownloaded false

Reply:
hmm, I got your point. But i don't think we save only downloaded songs into database. At least we don't save a song when pressing 'Download' button. So I thought all the songs that are visible to user are saved into database (after search for example). Is it true?

Reply:
For example I download 3 songs and in database I have only 3 songs
<img width="1175" alt="Screenshot 2022-10-24 at 17 59 08" src="https://user-images.githubusercontent.com/109308856/197558933-29e3565d-f0c5-4fb1-bf44-0da38af711db.png">

Reply:
try to just play 3 more songs without downloading them

Reply:
Oh, I see, I didn't know that songs which we played will save to database

Reply:
no problem. So looks like we have to set isDownlaoded for the currenly downloading song only. You can get song's id by accessing `downlaod.request.id` if i remember it correctly